### PR TITLE
Added an appdata.xml file for software gallery integration

### PR DESCRIPTION
**Pull request by Mailaender**
This file was created by @mariobl for Fedora. While incorporating it into openSUSE I thought it would be best to contribute it upstream. Adjusted the `updatecontact` tag accordingly.

https://people.freedesktop.org/~hughsient/appdata/ gives some explanation on what this is about.

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately there were the following style issues with your Pull Request:

 * Commit: 62c9c73adc616ecc44c5618bf2d917226babbe65
  * Your subject line is longer than 50 characters

Please follow [git commit message guidelines](http://chris.beams.io/posts/git-commit/) (you can use `git push -f` to update this PR).

---
This message was auto-generated by https://gitcop.com

**Comment by Mailaender** [comment]
Shortened the git commit message.

See https://build.opensuse.org/request/show/424652 for an integration test.

**Comment by vslavik** [comment]
> This file was created by @mariobl for Fedora. While incorporating it into openSUSE I thought it would be best to contribute it upstream.

Indeed, this is very helpful. Thanks!

> Shortened the git commit message.

That really is just a superficial part of GitCop’s message. The entire linked guide is worth following, not only in superficial things like short commit message or not ending the subject with period, but also writing explanatory commit messages — a brief note with the AppData spec link would probably be helpful.

File: poedit.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2015 Mario Blättermann <mariobl@fedoraproject.org> -->
+<application>
+  <id type="desktop">poedit.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>MIT</project_license>
+  <name>Poedit</name>
+  <name xml:lang="de">Poedit</name>
```
Comment (vslavik) [inline poedit.appdata.xml]: Is this necessary, e.g. for the description to be picked up? Seeing as the name is going to be “Poedit” in all languages, it seems pointless...

File: poedit.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      einen Quellcode-Parser. Es hilft beim Übersetzen von Anwendungen in
+      andere Sprachen.
+    </p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://upload.wikimedia.org/wikipedia/commons/c/c2/Poedit_1.8.1_en.png</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">http://poedit.net</url>
```
Comment (vslavik) [inline poedit.appdata.xml]: Please change to http**s**://poedit.net

**Comment by Mailaender** [comment]
Updated.

**Comment by vslavik** [comment]
> Updated.

… and merged. Thank you!

(Cherry-picked, actually, as 3e45b824d7a2e6c9f8f13dc23fe8849580cba6d6 onto the `maint-1.8` branch so that it is included in 1.8.10. Apparently, GitHub’s new rebasing feature is useless for actually rebasing onto another, non-trivially diverged branch.)

**Comment by mariobl** [comment]
Hi Vaclav,
nice to know that the appdata file will be part of the next release! But the version submitted by Matthias is not up-to-date referring to the latest changes in the Appstream project. See the appdata file in the Fedora Git repo [1].

Well, it is OK for the time being, but we need some other changes in the future. The copyright line currently refers to me:

You might add your own line:


Moreover, the German translations are obsolete once the appdata file is integrated into the translation workflow for Poedit. See [2] for more information. As far as I can see, it seems to be quite simple using an *xml.in  file instead of *xml.

[1] http://pkgs.fedoraproject.org/cgit/rpms/poedit.git/plain/poedit.appdata.xml
[2] https://people.freedesktop.org/~hughsient/appdata/

**Comment by Mailaender** [comment]
Yes, I also thought about making this localisable from the beginning, but I am a bit inexperienced with it. See https://github.com/PintaProject/Pinta/pull/121 and https://github.com/PintaProject/Pinta/pull/122 for my attempts on another project.

**Comment by vslavik** [comment]
> Well, it is OK for the time being, but we need some other changes in the future. 

Can you please make a PR then? For the length of your comment, it’s unfortunately very light on specifics, and all I get from it is “all you did is wrong”, but not *how* or how to fix it.

> The copyright line currently refers to me:
> You might add your own line:

Damn, didn’t catch that. I’d actually much prefer to remove it altogether if you don’t mind. It’s schizophrenic, you see: on this line you claim copyright, but just 3 lines under it, you explicitly put the file into public domain (aka CC0). You can’t have it both ways. IMHO the latter makes more sense given how the file’s content (perhaps with the exception of the description) doesn’t meet the creativity requirement for copyright protection in the first place.

**Comment by Mailaender** [comment]
Filed https://github.com/vslavik/poedit/pull/287 myself as I can see what he changed. Sorry, I somehow must have picked an older version on Fedora git. I had some scruples to remove the `(C)` as the translation was something that has some threshold of originality.

**Comment by mariobl** [comment]
@mailaender, the time you picked up the appdata file from Fedora git it was up-to-date, I've updated it some days later.

**Pull request closed without merging**